src/commands/synthetics/__tests__/build-and-test.test.ts

```diff
@@ -0,0 +1,55 @@
+import {once} from 'events'
+import * as http from 'http'
+
+import {spawnBuildPluginDevServer} from '../build-and-test'
+
+import {mockReporter} from './fixtures'
+
+// if BUILD_PLUGINS_S8S_PORT is defined, this command does nothing forever, otherwise, it exits immediately
+const MOCKED_BUILD_COMMAND = (port: number) => `[ "$BUILD_PLUGINS_S8S_PORT" = "${port}" ] && tail -f /dev/null`
+// this command simulates a build command without the build plugin configured
+const MOCKED_BUILD_COMMAND_NOT_CONFIGURED = 'echo "build successful"'
```

Comment:
Can you avoid really running any bash command in unit tests? I'm working on running all our unit tests on Windows, where `tail` and `echo` won't be available.

Reply:
You could maybe mock `spawn` altogether?

Reply:
I went with an agnostic implementation leveraging the node executable running the tests.
I'd rather avoid mocking `spawn` to make sure the env variable is propagated, and the child process is correctly executed and being killed, as it'll probably be the main source of issue in this implementation, I reckon.

Reply:
> I'd rather avoid mocking spawn to make sure the env variable is propagated, and the child process is correctly executed and being killed

How do you make sure of this here?
I feel like these are not currently assessed in the tests.